core/config-schema/schema.json

```diff
@@ -1618,6 +1639,46 @@
       },
       "additionalProperties": false
     },
+    "NsisConfig": {
+      "description": "Configuration for the Installer bundle using NSIS.",
+      "type": "object",
+      "properties": {
+        "license": {
+          "description": "The path to the license file to render on the installer.",
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "headerImage": {
+          "description": "The path to a bitmap file to display on the header of installers pages.\n\nThe recommended dimensions are 150pxx57px.",
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "sidebarImage": {
+          "description": "The path to a bitmap file for the Welcome page and the Finish page.\n\nThe recommended dimensions are 164pxx314px.",
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "installerIcon": {
+          "description": "The path to an icon file used as the installer icon.",
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "perMachine": {
+          "description": "Whether the installation will be for all users or just the current user.",
+          "default": false,
+          "type": "boolean"
+        }
```

Comment:
Ah, just saw this!

What about having a third state, and making the value more understandable without a boolean value?
- `"all_users"` - All users
- `"current_user"` - Current user
- Accepts object (optional)
  - `"recommended_option"` or `"highlighted_option"`: `"all_users" | "current_user"`

The third object will make both options visible and recommend an item, instead of forcing to install the app to a pre-selected target.

Reply:
The thing is if your app logic depends on writing some files next to the exe, you can't use `all_users` so I wanted the system-wide option to be explicitly opted-in, otherwise your users might report unexpected errors because they installed it to a location that doesn't have write access. Also, that the installer will have to be launched as administrator even if the user plan to use `current_user`.

Reply:
I think I might change this option to be

* `"all_users"` - All users
* `"current_user"` - Current user
* `"both"`

Although this might complicate the logic of the installer. Thanks for the feedback.

Reply:
> > Also, that the installer will have to be launched as administrator even if the user plan to use current_user
> 
> This kind of concerns me. Some users may not have an admin privilege and still want to install the app. I'm assuming there's no way around that as well?

This will only be a concern if you use `all_users` or `both` option when building the installer.

Reply:
The only way I found is to write 3 installers, one that prompts users to choose `current_user` or `all_users` and depending on that, launches one of the other two installers. Which is messy and will pollute the installer size.